mypy/types.py

```diff
@@ -384,6 +384,7 @@ class Instance(Type):
 
     def __init__(self, typ: Optional[mypy.nodes.TypeInfo], args: List[Type],
                  line: int = -1, column: int = -1, erased: bool = False) -> None:
+        assert(typ is None or typ.fullname() not in ["builtins.Any", "typing.Any"])
```

Comment:
This test is not right -- `typ` should never be `None`. What's the purpose of this?